File: Colas/Arreglos/cola.py
MAXSIZE = 5
queue = [0] * MAXSIZE
front = -1
rear = -1

def insertar():
    global front, rear
    if rear == MAXSIZE - 1:
        print("\nOVERFLOW")
        return
    
    elemento = int(input("\nIngrese el elemento: "))
    
    if front == -1 and rear == -1:
        front = rear = 0
    else:
        rear += 1
    queue[rear] = elemento
    print("\nElemento insertado correctamente.\n")

def eliminar():
    global front, rear
    if front == -1 or front > rear:
        print("\nUnderflow")
        return
    
    elemento = queue[front]
    if front == rear:
        front = rear = -1
    else:
        front += 1
    print(f"\nElemento eliminado: {elemento}\n")

def mostrar():
    if rear == -1 or front == -1 or front > rear:
        print("\nLa cola esta vacia.\n")
    else:
        print("\nElementos en la cola:")
        for i in range(front, rear + 1):
            print(queue[i])

def main():
    opcion = 0
    while opcion != 4:
        print("\n==========================MENU PRINCIPAL==========================")
        print("====================================================================")
        print("1. Insertar\n2. Eliminar\n3. Mostrar\n4. Salir")
        opcion = int(input("Ingrese su opcion: "))
        
        if opcion == 1:
            insertar()
        elif opcion == 2:
            eliminar()
        elif opcion == 3:
            mostrar()
        elif opcion == 4:
            print("\nSaliendo del programa...\n")
        else:
            print("\nOpcion invalida. Intente nuevamente\n")

if __name__ == "__main__":
    main()

File: Colas/Arreglos/javaScript/cola.js
const readline = require('readline');

const MAXSIZE = 5;
let queue = new Array(MAXSIZE);
let front = -1, rear = -1;

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function insertar() {
    if (rear === MAXSIZE - 1) {
        console.log("\nOVERFLOW");
        mostrarMenu();
        return;
    }

    rl.question("\nIngrese el elemento: ", (input) => {
        const elemento = parseInt(input);
        
        if (front === -1 && rear === -1) {
            front = rear = 0;
        } else {
            rear++;
        }
        queue[rear] = elemento;
        console.log("\nElemento insertado correctamente.\n");
        mostrarMenu();
    });
}

function eliminar() {
    if (front === -1 || front > rear) {
        console.log("\nUnderflow");
        mostrarMenu();
        return;
    }

    const elemento = queue[front];
    if (front === rear) {
        front = rear = -1;
    } else {
        front++;
    }
    console.log(`\nElemento eliminado: ${elemento}\n`);
    mostrarMenu();
}

function mostrar() {
    if (rear === -1 || front === -1 || front > rear) {
        console.log("\nLa cola esta vacia.\n");
    } else {
        console.log("\nElementos en la cola:");
        for (let i = front; i <= rear; i++) {
            console.log(queue[i]);
        }
    }
    mostrarMenu();
}

function mostrarMenu() {
    console.log("\n==========================MENU PRINCIPAL==========================");
    console.log("====================================================================");
    console.log("1. Insertar\n2. Eliminar\n3. Mostrar\n4. Salir");
    
    rl.question("Ingrese su opcion: ", (opcion) => {
        switch (opcion) {
            case "1": insertar(); break;
            case "2": eliminar(); break;
            case "3": mostrar(); break;
            case "4": 
                console.log("\nSaliendo del programa...\n");
                rl.close();
                break;
            default: 
                console.log("\nOpcion invalida. Intente nuevamente\n");
                mostrarMenu();
                break;
        }
    });
}

// Iniciar el programa
mostrarMenu();